x-pack/platform/plugins/shared/onechat/public/application/components/tools/execute/test_tools.tsx

```
@@ -0,0 +1,217 @@
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiFormRow,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { useForm, FormProvider, Controller } from 'react-hook-form';
+import type { ToolDefinition } from '@kbn/onechat-common';
+import { i18n } from '@kbn/i18n';
+import { CodeEditor } from '@kbn/code-editor';
+import { useExecuteTool } from '../../../hooks/tools/use_execute_tools';
+import type { ExecuteToolResponse } from '../../../../../common/http_api/tools';
+
+interface OnechatTestToolFlyoutProps {
+  isOpen: boolean;
+  isLoading?: boolean;
+  tool?: ToolDefinition;
+  onClose: () => void;
+}
+
+const getParameters = (tool: ToolDefinition | undefined) => {
+  if (!tool) return [];
+
+  const fields: Array<{ name: string; label: string; value: string; type: string }> = [];
+  if (tool.configuration && tool.configuration.params) {
+    const params = tool.configuration.params as Record<string, any>;
+    Object.entries(params).forEach(([paramName, paramConfig]) => {
+      fields.push({
+        name: `${paramName}`,
+        label: `${paramName}`,
+        value: '',
+        type: paramConfig.type || 'text',
+      });
+    });
+  }
+
+  return fields;
+};
+
+export const OnechatTestFlyout: React.FC<OnechatTestToolFlyoutProps> = ({
+  isOpen,
+  isLoading,
+  tool,
+  onClose,
+}) => {
+  const [response, setResponse] = useState<string>('{}');
+
+  const form = useForm({
+    mode: 'onChange',
+  });
+  const {
+    handleSubmit,
+    formState: { errors },
+  } = form;
+
+  const { executeTool, isLoading: isExecuting } = useExecuteTool({
+    onSuccess: (data: ExecuteToolResponse) => {
+      setResponse(JSON.stringify(data, null, 2));
+    },
+    onError: (error: Error) => {
+      setResponse(JSON.stringify({ error: error.message }, null, 2));
+    },
+  });
+
+  const onSubmit = async (formData: Record<string, any>) => {
+    if (!tool?.id) return;
```

This is silent failure, user won't know why we failed?

---

This shouldn't even really be possible to submit without a tool id since the flyout can only be shown after saving if creating a tool, so I could just take this out?